packages/dci-api/src/validations.ts

```diff
@@ -87,58 +87,66 @@ const reference = (value: ZodType = z.string()) =>
     value
   })
 
-const attributeValue = z.string().or(z.number()).or(z.boolean())
+const commonSearchCriteria = z.object({
+  version: version.optional(),
+  reg_type: reference().optional(),
+  reg_event_type: reference(eventTypes),
+  result_record_type: reference(),
+  sort: z.array(searchSort).optional(),
+  pagination: paginationRequest.optional(),
+  consent: consent.optional(),
+  authorize: authorize.optional()
+})
+
+const identifier = z.enum(['BRN', 'DRN', 'MRN', 'OPENCRVS_RECORD_ID', 'NID'])
 
 const identifierTypeValue = z.object({
-  identifier_type: reference(),
-  identifier_value: attributeValue
+  identifier_type: reference(identifier),
+  identifier_value: z.string()
 })
 
-const identifierTypeQuery = z.object({
-  query_type: z.literal('idtype-value'),
-  query: identifierTypeValue
-})
+const identifierTypeQuery = commonSearchCriteria.and(
+  z.object({
+    query_type: z.literal('idtype-value'),
+    query: identifierTypeValue
+  })
+)
+
+const expressionCondition = z.enum(['and'])
 
-const expressionCondition = z.enum(['and', 'or', 'not'])
+const expression = z.enum(['gt', 'lt', 'eq', 'ge', 'le'])
 
-const expressionOperator = z.enum(['gt', 'lt', 'eq', 'ge', 'le', 'in'])
+const expressionSupportedFields = z.enum(['birthdate'])
 
 const expressionPredicate = z.object({
-  attribute_name: z.string(),
-  operator: expressionOperator,
-  attribute_value: attributeValue
+  attribute_name: expressionSupportedFields,
+  operator: expression,
+  attribute_value: z.coerce.date()
 })
 
-const predicateQuery = z.object({
-  query_type: z.literal('predicate'),
-  query: z.array(
-    z.object({
-      seq_num: z.number().optional(),
-      expression1: expressionPredicate,
-      condition: expressionCondition.optional(),
-      expression2: expressionPredicate.optional()
-    })
-  )
-})
+const predicateQuery = commonSearchCriteria.and(
+  z.object({
+    query_type: z.literal('predicate'),
+    query: z.array(
+      z.object({
+        seq_num: z.number().optional(),
+        expression1: expressionPredicate,
+        condition: expressionCondition.optional(),
+        expression2: expressionPredicate
+      })
```

Comment:
if `condition` is optional then shouldn't `expression2` be so as well?

Reply:
Hmm solid point.. I should probably make it
```
condition: z.literal('and'),
expression2: ...
```
so it's obvious that they both are required in our schema 🤔

Reply:
yeah that would be better